src/main/java/taxi/service/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,25 @@
+package taxi.service;
+
+import java.util.Optional;
+import taxi.lib.Injector;
+import taxi.lib.Service;
+import taxi.model.Driver;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    private static final Injector injector = Injector.getInstance("taxi");
+    private static final DriverService driverService =
+            (DriverService) injector.getInstance(DriverService.class);
+
+    @Override
+    public Driver login(String driverName, String driverPassword) {
+        Optional<Driver> driver = driverService.findByLogin(driverName);
+        if (driver.isEmpty()) {
+            throw new RuntimeException("login or password was incorrect");
+        }
+        if (driver.get().getPassword().equals(driverPassword)) {
+            return driver.get();
+        }
```

Comment:
[Checklist](https://mate-academy.github.io/jv-program-common-mistakes/java-web/web-security/java-web-security)
Both in AuthenticationServiceImpl and in AuthenticationFilter, use only one if block if possible.
**Hint: it's possible**